hack/lib/util/environment.sh

```diff
@@ -148,7 +148,17 @@ function os::util::environment::setup_tmpdir_vars() {
     HOME="${FAKE_HOME_DIR}"
     export HOME
 
-    mkdir -p  "${BASETMPDIR}" "${LOG_DIR}" "${VOLUME_DIR}" "${ARTIFACT_DIR}" "${HOME}"
+    # ensure that the directories are clean
+    for target in $( ${USE_SUDO:+sudo} findmnt -output TARGET --list ); do
+        if grep -q "${BASETMPDIR}" <<< "${target}"; then
+            ${USE_SUDO:+sudo} umount "${target}"
```

Comment:
I think the original code was written "as a user would do it" in that when we were cleaning things up we would try to delete everything and if that failed it meant we had not unmounted something, so we did that and tried the delete again. I feel this is a better approach since we _always_ want to unmount stuff if it is mounted, as this will _always_ fail the `rm -rf`. 

I tried to find a way to get `findmnt` to output mounts with targets below a directory, but `--target` instead searches up the tree. `$BASETMPDIR` like you say will be long enough to be fine, I think.